crates/ef-testing/src/sequencer/mod.rs

```diff
@@ -0,0 +1,137 @@
+use blockifier::state::errors::StateError;
+use blockifier::state::state_api::{State as BlockifierState, StateResult};
+use cairo_vm::types::errors::program_errors::ProgramError;
+use reth_primitives::Address;
+use sequencer::sequencer::Sequencer;
+use sequencer::state::State;
+use starknet::core::utils::get_contract_address;
+
+use self::constants::{
+    BLOCK_CONTEXT, CONTRACT_ACCOUNT_CLASS, CONTRACT_ACCOUNT_CLASS_HASH, EOA_CLASS, EOA_CLASS_HASH,
+    FEE_TOKEN_ADDRESS, KAKAROT_ADDRESS, KAKAROT_CLASS, KAKAROT_CLASS_HASH, KAKAROT_OWNER_ADDRESS,
+    PROXY_CLASS, PROXY_CLASS_HASH,
+};
+use self::types::FeltSequencer;
+use self::utils::{
+    class_hash_to_starkfelt, contract_address_to_starkfelt, get_storage_var_address,
+};
+
+pub(crate) struct KakarotSequencer(Sequencer<State>);
+
+#[allow(dead_code)]
+impl KakarotSequencer {
+    pub fn new(state: State) -> Self {
+        let sequencer = Sequencer::new(BLOCK_CONTEXT.clone(), state);
+        Self(sequencer)
+    }
+
+    pub fn initialize(mut self) -> StateResult<Self> {
+        let mut storage = vec![];
+
+        // Initialize the kakarot owner.
+        let kakarot_owner_storage = (
+            get_storage_var_address("Ownable_owner", &[]).unwrap(), // safe unwrap: var is ASCII
+            contract_address_to_starkfelt(&KAKAROT_OWNER_ADDRESS),
+        );
+        storage.push(kakarot_owner_storage);
+
+        // Initialize the kakarot fee token address.
+        let kakarot_fee_token_storage = (
+            get_storage_var_address("native_token_address", &[]).unwrap(), // safe unwrap: var is ASCII
+            contract_address_to_starkfelt(&FEE_TOKEN_ADDRESS),
+        );
+        storage.push(kakarot_fee_token_storage);
+
+        // Initialize the kakarot various class hashes.
+        let kakarot_class_hashes = &mut vec![
+            (
+                get_storage_var_address("contract_account_class_hash", &[]).unwrap(),
+                class_hash_to_starkfelt(&CONTRACT_ACCOUNT_CLASS_HASH),
+            ),
+            (
+                get_storage_var_address("externally_owned_account_class_hash", &[]).unwrap(),
+                class_hash_to_starkfelt(&EOA_CLASS_HASH),
+            ),
+            (
+                get_storage_var_address("account_proxy_class_hash", &[]).unwrap(),
+                class_hash_to_starkfelt(&PROXY_CLASS_HASH),
+            ),
+        ];
+        storage.append(kakarot_class_hashes);
+
+        // Write all the storage vars to the sequencer state.
+        for (k, v) in storage {
+            (&mut self.0.state).set_storage_at(*KAKAROT_ADDRESS, k, v);
+        }
+
+        // Write the kakarot class and class hash.
+        (&mut self.0.state).set_class_hash_at(*KAKAROT_ADDRESS, *KAKAROT_CLASS_HASH)?;
+        (&mut self.0.state).set_contract_class(
+            &KAKAROT_CLASS_HASH,
+            ContractClass::V0(ContractClassV0::try_from_json_string(
+                &serde_json::to_string(&*KAKAROT_CLASS)
```

Comment:
Issue is that `KAKAROT_CLASS` is a type from starknet rs, and is useful because it has a function `class_hash()` which allows to compute the class hash of the class easily. However, this function is not present on the blockifier contract class type, that is needed to set the contract class in storage. 

I could implement a conversion function, but given the fact that this setup should only be used to make a dump and is already quite fast (~ 10 seconds), I did not see the need for this.